lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1650,9 +1666,25 @@ def cmd_sessions(*args)
         end
       end
     when 'killall'
-      print_status("Killing all sessions...")
-      framework.sessions.each_sorted.reverse_each do |s|
-        session = framework.sessions.get(s)
+      if search_term
+        matching_sessions = get_matching_sessions(search_term)
+        if matching_sessions.empty?
+          print_status("No matching sessions.") 
+          return
+        end
+        print_status("Killing matching sessions...")
+        print_line
+        print(Serializer::ReadableText.dump_sessions(framework, show_active: show_active, show_inactive: show_inactive, show_extended: show_extended, verbose: verbose, session_ids: matching_sessions.keys()))
+        print_line
+      else
+        matching_sessions = framework.sessions
+        print_status("Killing all sessions...")
+      end
+      matching_sessions.each do |session_id, session|
+        unless matching_sessions.nil? || matching_sessions.include?(session_id)
```

Comment:
if `matching_sessions` can be nil here, which is what you're checking for, it would throw an exception on `matching_sessions.each` first